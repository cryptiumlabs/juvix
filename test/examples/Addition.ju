open Prelude

let nat = %Michelson.nat

let add = %Michelson.add

let pair = %Michelson.pair

type Storage  = {
  value  : nat,
  valu : nat
}

sig foo : nat -> storage -> storage
let foo va {value, valu} =
  {value = add valu va, valu = add value va}

sig add-storage : storage -> nat
let add-storage {value, valu} =
  add value valu

-- Default entry point
sig default : nat -> storage -> storage
let default addnat storage = pair (add-storage storage) (foo input storage)