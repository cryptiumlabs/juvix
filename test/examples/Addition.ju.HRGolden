NoHeader
    [1m[96m[[0m ModuleOpen
        [1m[93m([0m Open [35m([0m [1m[97m"[0m[1m[94mPrelude[0m[1m[97m"[0m :| [36m[[0m[36m][0m [35m)[0m [1m[93m)[0m
    [1m[96m,[0m Function
        [1m[93m([0m Func
            [35m([0m Like
                [36m{[0m functionLikedName = [1m[97m"[0m[1m[94mnat[0m[1m[97m"[0m
                [36m,[0m functionLikeArgs = [33m[[0m[33m][0m
                [36m,[0m functionLikeBody = Body
                    [33m([0m Primitive
                        [1m[35m([0m Prim
                            [1m[36m([0m [1m[97m"[0m[1m[94mMichelson[0m[1m[97m"[0m :| [1m[33m[[0m [1m[97m"[0m[1m[94mnat[0m[1m[97m"[0m [1m[33m][0m [1m[36m)[0m
                        [1m[35m)[0m
                    [33m)[0m
                [36m}[0m
            [35m)[0m
        [1m[93m)[0m
    [1m[96m,[0m Function
        [1m[93m([0m Func
            [35m([0m Like
                [36m{[0m functionLikedName = [1m[97m"[0m[1m[94madd[0m[1m[97m"[0m
                [36m,[0m functionLikeArgs = [33m[[0m[33m][0m
                [36m,[0m functionLikeBody = Body
                    [33m([0m Primitive
                        [1m[35m([0m Prim
                            [1m[36m([0m [1m[97m"[0m[1m[94mMichelson[0m[1m[97m"[0m :| [1m[33m[[0m [1m[97m"[0m[1m[94madd[0m[1m[97m"[0m [1m[33m][0m [1m[36m)[0m
                        [1m[35m)[0m
                    [33m)[0m
                [36m}[0m
            [35m)[0m
        [1m[93m)[0m
    [1m[96m,[0m Function
        [1m[93m([0m Func
            [35m([0m Like
                [36m{[0m functionLikedName = [1m[97m"[0m[1m[94mpair[0m[1m[97m"[0m
                [36m,[0m functionLikeArgs = [33m[[0m[33m][0m
                [36m,[0m functionLikeBody = Body
                    [33m([0m Primitive
                        [1m[35m([0m Prim
                            [1m[36m([0m [1m[97m"[0m[1m[94mMichelson[0m[1m[97m"[0m :| [1m[33m[[0m [1m[97m"[0m[1m[94mpair[0m[1m[97m"[0m [1m[33m][0m [1m[36m)[0m
                        [1m[35m)[0m
                    [33m)[0m
                [36m}[0m
            [35m)[0m
        [1m[93m)[0m
    [1m[96m,[0m Type
        [1m[93m([0m Typ
            [35m{[0m typeUsage = Nothing
            [35m,[0m typeName' = [1m[97m"[0m[1m[94mStorage[0m[1m[97m"[0m
            [35m,[0m typeArgs = [36m[[0m[36m][0m
            [35m,[0m typeForm = NonArrowed
                [36m{[0m dataAdt = Product
                    [33m([0m Record
                        [1m[35m([0m Record''
                            [1m[36m{[0m recordFields = NameType'
                                [1m[33m{[0m nameTypeSignature = Name [95m([0m [1m[97m"[0m[1m[94mnat[0m[1m[97m"[0m :| [96m[[0m[96m][0m [95m)[0m
                                [1m[33m,[0m nameTypeName = Concrete [1m[97m"[0m[1m[94mvalue[0m[1m[97m"[0m
                                [1m[33m}[0m :|
                                [1m[33m[[0m NameType'
                                    [95m{[0m nameTypeSignature = Name [96m([0m [1m[97m"[0m[1m[94mnat[0m[1m[97m"[0m :| [93m[[0m[93m][0m [96m)[0m
                                    [95m,[0m nameTypeName = Concrete [1m[97m"[0m[1m[94mvalu[0m[1m[97m"[0m
                                    [95m}[0m
                                [1m[33m][0m
                            [1m[36m,[0m recordFamilySignature = Nothing
                            [1m[36m}[0m
                        [1m[35m)[0m
                    [33m)[0m
                [36m}[0m
            [35m}[0m
        [1m[93m)[0m
    [1m[96m,[0m Signature
        [1m[93m([0m Sig
            [35m{[0m signatureName = [1m[97m"[0m[1m[94mfoo[0m[1m[97m"[0m
            [35m,[0m signatureUsage = Nothing
            [35m,[0m signatureArrowType = Infix
                [36m([0m Inf
                    [33m{[0m infixLeft = Name [1m[35m([0m [1m[97m"[0m[1m[94mnat[0m[1m[97m"[0m :| [1m[36m[[0m[1m[36m][0m [1m[35m)[0m
                    [33m,[0m infixOp = [1m[97m"[0m[1m[94m->[0m[1m[97m"[0m :| [1m[35m[[0m[1m[35m][0m
                    [33m,[0m infixRight = Infix
                        [1m[35m([0m Inf
                            [1m[36m{[0m infixLeft = Name [1m[33m([0m [1m[97m"[0m[1m[94mstorage[0m[1m[97m"[0m :| [95m[[0m[95m][0m [1m[33m)[0m
                            [1m[36m,[0m infixOp = [1m[97m"[0m[1m[94m->[0m[1m[97m"[0m :| [1m[33m[[0m[1m[33m][0m
                            [1m[36m,[0m infixRight = Name [1m[33m([0m [1m[97m"[0m[1m[94mstorage[0m[1m[97m"[0m :| [95m[[0m[95m][0m [1m[33m)[0m
                            [1m[36m}[0m
                        [1m[35m)[0m
                    [33m}[0m
                [36m)[0m
            [35m,[0m signatureConstraints = [36m[[0m[36m][0m
            [35m}[0m
        [1m[93m)[0m
    [1m[96m,[0m Function
        [1m[93m([0m Func
            [35m([0m Like
                [36m{[0m functionLikedName = [1m[97m"[0m[1m[94mfoo[0m[1m[97m"[0m
                [36m,[0m functionLikeArgs =
                    [33m[[0m ConcreteA
                        [1m[35m([0m MatchLogic
                            [1m[36m{[0m matchLogicContents = MatchName [1m[97m"[0m[1m[94mva[0m[1m[97m"[0m
                            [1m[36m,[0m matchLogicNamed = Nothing
                            [1m[36m}[0m
                        [1m[35m)[0m
                    [33m,[0m ConcreteA
                        [1m[35m([0m MatchLogic
                            [1m[36m{[0m matchLogicContents = MatchRecord
                                [1m[33m([0m Punned [95m([0m [1m[97m"[0m[1m[94mvalue[0m[1m[97m"[0m :| [96m[[0m[96m][0m [95m)[0m :|
                                    [95m[[0m Punned [96m([0m [1m[97m"[0m[1m[94mvalu[0m[1m[97m"[0m :| [93m[[0m[93m][0m [96m)[0m [95m][0m
                                [1m[33m)[0m
                            [1m[36m,[0m matchLogicNamed = Nothing
                            [1m[36m}[0m
                        [1m[35m)[0m
                    [33m][0m
                [36m,[0m functionLikeBody = Body
                    [33m([0m ExpRecord
                        [1m[35m([0m ExpressionRecord
                            [1m[36m{[0m expRecordFields = NonPunned [1m[33m([0m [1m[97m"[0m[1m[94mvalue[0m[1m[97m"[0m :| [95m[[0m[95m][0m [1m[33m)[0m
                                [1m[33m([0m Application
                                    [95m([0m App
                                        [96m{[0m applicationName = Name [93m([0m [1m[97m"[0m[1m[94madd[0m[1m[97m"[0m :| [1m[95m[[0m[1m[95m][0m [93m)[0m
                                        [96m,[0m applicationArgs = Name [93m([0m [1m[97m"[0m[1m[94mvalu[0m[1m[97m"[0m :| [1m[95m[[0m[1m[95m][0m [93m)[0m :|
                                            [93m[[0m Name [1m[95m([0m [1m[97m"[0m[1m[94mva[0m[1m[97m"[0m :| [1m[96m[[0m[1m[96m][0m [1m[95m)[0m [93m][0m
                                        [96m}[0m
                                    [95m)[0m
                                [1m[33m)[0m :|
                                [1m[33m[[0m NonPunned [95m([0m [1m[97m"[0m[1m[94mvalu[0m[1m[97m"[0m :| [96m[[0m[96m][0m [95m)[0m
                                    [95m([0m Application
                                        [96m([0m App
                                            [93m{[0m applicationName = Name [1m[95m([0m [1m[97m"[0m[1m[94madd[0m[1m[97m"[0m :| [1m[96m[[0m[1m[96m][0m [1m[95m)[0m
                                            [93m,[0m applicationArgs = Name [1m[95m([0m [1m[97m"[0m[1m[94mvalue[0m[1m[97m"[0m :| [1m[96m[[0m[1m[96m][0m [1m[95m)[0m :|
                                                [1m[95m[[0m Name [1m[96m([0m [1m[97m"[0m[1m[94mva[0m[1m[97m"[0m :| [1m[93m[[0m[1m[93m][0m [1m[96m)[0m [1m[95m][0m
                                            [93m}[0m
                                        [96m)[0m
                                    [95m)[0m
                                [1m[33m][0m
                            [1m[36m}[0m
                        [1m[35m)[0m
                    [33m)[0m
                [36m}[0m
            [35m)[0m
        [1m[93m)[0m
    [1m[96m,[0m Signature
        [1m[93m([0m Sig
            [35m{[0m signatureName = [1m[97m"[0m[1m[94madd-storage[0m[1m[97m"[0m
            [35m,[0m signatureUsage = Nothing
            [35m,[0m signatureArrowType = Infix
                [36m([0m Inf
                    [33m{[0m infixLeft = Name [1m[35m([0m [1m[97m"[0m[1m[94mstorage[0m[1m[97m"[0m :| [1m[36m[[0m[1m[36m][0m [1m[35m)[0m
                    [33m,[0m infixOp = [1m[97m"[0m[1m[94m->[0m[1m[97m"[0m :| [1m[35m[[0m[1m[35m][0m
                    [33m,[0m infixRight = Name [1m[35m([0m [1m[97m"[0m[1m[94mnat[0m[1m[97m"[0m :| [1m[36m[[0m[1m[36m][0m [1m[35m)[0m
                    [33m}[0m
                [36m)[0m
            [35m,[0m signatureConstraints = [36m[[0m[36m][0m
            [35m}[0m
        [1m[93m)[0m
    [1m[96m,[0m Function
        [1m[93m([0m Func
            [35m([0m Like
                [36m{[0m functionLikedName = [1m[97m"[0m[1m[94madd-storage[0m[1m[97m"[0m
                [36m,[0m functionLikeArgs =
                    [33m[[0m ConcreteA
                        [1m[35m([0m MatchLogic
                            [1m[36m{[0m matchLogicContents = MatchRecord
                                [1m[33m([0m Punned [95m([0m [1m[97m"[0m[1m[94mvalue[0m[1m[97m"[0m :| [96m[[0m[96m][0m [95m)[0m :|
                                    [95m[[0m Punned [96m([0m [1m[97m"[0m[1m[94mvalu[0m[1m[97m"[0m :| [93m[[0m[93m][0m [96m)[0m [95m][0m
                                [1m[33m)[0m
                            [1m[36m,[0m matchLogicNamed = Nothing
                            [1m[36m}[0m
                        [1m[35m)[0m
                    [33m][0m
                [36m,[0m functionLikeBody = Body
                    [33m([0m Application
                        [1m[35m([0m App
                            [1m[36m{[0m applicationName = Name [1m[33m([0m [1m[97m"[0m[1m[94madd[0m[1m[97m"[0m :| [95m[[0m[95m][0m [1m[33m)[0m
                            [1m[36m,[0m applicationArgs = Name [1m[33m([0m [1m[97m"[0m[1m[94mvalue[0m[1m[97m"[0m :| [95m[[0m[95m][0m [1m[33m)[0m :|
                                [1m[33m[[0m Name [95m([0m [1m[97m"[0m[1m[94mvalu[0m[1m[97m"[0m :| [96m[[0m[96m][0m [95m)[0m [1m[33m][0m
                            [1m[36m}[0m
                        [1m[35m)[0m
                    [33m)[0m
                [36m}[0m
            [35m)[0m
        [1m[93m)[0m
    [1m[96m,[0m Signature
        [1m[93m([0m Sig
            [35m{[0m signatureName = [1m[97m"[0m[1m[94mdefault[0m[1m[97m"[0m
            [35m,[0m signatureUsage = Nothing
            [35m,[0m signatureArrowType = Infix
                [36m([0m Inf
                    [33m{[0m infixLeft = Name [1m[35m([0m [1m[97m"[0m[1m[94mnat[0m[1m[97m"[0m :| [1m[36m[[0m[1m[36m][0m [1m[35m)[0m
                    [33m,[0m infixOp = [1m[97m"[0m[1m[94m->[0m[1m[97m"[0m :| [1m[35m[[0m[1m[35m][0m
                    [33m,[0m infixRight = Infix
                        [1m[35m([0m Inf
                            [1m[36m{[0m infixLeft = Name [1m[33m([0m [1m[97m"[0m[1m[94mstorage[0m[1m[97m"[0m :| [95m[[0m[95m][0m [1m[33m)[0m
                            [1m[36m,[0m infixOp = [1m[97m"[0m[1m[94m->[0m[1m[97m"[0m :| [1m[33m[[0m[1m[33m][0m
                            [1m[36m,[0m infixRight = Name [1m[33m([0m [1m[97m"[0m[1m[94mstorage[0m[1m[97m"[0m :| [95m[[0m[95m][0m [1m[33m)[0m
                            [1m[36m}[0m
                        [1m[35m)[0m
                    [33m}[0m
                [36m)[0m
            [35m,[0m signatureConstraints = [36m[[0m[36m][0m
            [35m}[0m
        [1m[93m)[0m
    [1m[96m,[0m Function
        [1m[93m([0m Func
            [35m([0m Like
                [36m{[0m functionLikedName = [1m[97m"[0m[1m[94mdefault[0m[1m[97m"[0m
                [36m,[0m functionLikeArgs =
                    [33m[[0m ConcreteA
                        [1m[35m([0m MatchLogic
                            [1m[36m{[0m matchLogicContents = MatchName [1m[97m"[0m[1m[94maddnat[0m[1m[97m"[0m
                            [1m[36m,[0m matchLogicNamed = Nothing
                            [1m[36m}[0m
                        [1m[35m)[0m
                    [33m,[0m ConcreteA
                        [1m[35m([0m MatchLogic
                            [1m[36m{[0m matchLogicContents = MatchName [1m[97m"[0m[1m[94mstorage[0m[1m[97m"[0m
                            [1m[36m,[0m matchLogicNamed = Nothing
                            [1m[36m}[0m
                        [1m[35m)[0m
                    [33m][0m
                [36m,[0m functionLikeBody = Body
                    [33m([0m Application
                        [1m[35m([0m App
                            [1m[36m{[0m applicationName = Name [1m[33m([0m [1m[97m"[0m[1m[94mpair[0m[1m[97m"[0m :| [95m[[0m[95m][0m [1m[33m)[0m
                            [1m[36m,[0m applicationArgs = Parened
                                [1m[33m([0m Application
                                    [95m([0m App
                                        [96m{[0m applicationName = Name [93m([0m [1m[97m"[0m[1m[94madd-storage[0m[1m[97m"[0m :| [1m[95m[[0m[1m[95m][0m [93m)[0m
                                        [96m,[0m applicationArgs = Name [93m([0m [1m[97m"[0m[1m[94mstorage[0m[1m[97m"[0m :| [1m[95m[[0m[1m[95m][0m [93m)[0m :| [93m[[0m[93m][0m
                                        [96m}[0m
                                    [95m)[0m
                                [1m[33m)[0m :|
                                [1m[33m[[0m Parened
                                    [95m([0m Application
                                        [96m([0m App
                                            [93m{[0m applicationName = Name [1m[95m([0m [1m[97m"[0m[1m[94mfoo[0m[1m[97m"[0m :| [1m[96m[[0m[1m[96m][0m [1m[95m)[0m
                                            [93m,[0m applicationArgs = Name [1m[95m([0m [1m[97m"[0m[1m[94minput[0m[1m[97m"[0m :| [1m[96m[[0m[1m[96m][0m [1m[95m)[0m :|
                                                [1m[95m[[0m Name [1m[96m([0m [1m[97m"[0m[1m[94mstorage[0m[1m[97m"[0m :| [1m[93m[[0m[1m[93m][0m [1m[96m)[0m [1m[95m][0m
                                            [93m}[0m
                                        [96m)[0m
                                    [95m)[0m
                                [1m[33m][0m
                            [1m[36m}[0m
                        [1m[35m)[0m
                    [33m)[0m
                [36m}[0m
            [35m)[0m
        [1m[93m)[0m
    [1m[96m][0m