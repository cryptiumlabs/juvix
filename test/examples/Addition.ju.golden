
  [ Function'
      ( Func'
          ( Like'
              { functionLikedName = "nat",
                functionLikeArgs = [],
                functionLikeBody = Body' (Primitive' (Prim' ("Michelson" :| ["nat"]) ()) ()) (),
                annLike = ()
              }
          )
          ()
      )
      (),
    Function'
      ( Func'
          ( Like'
              { functionLikedName = "add",
                functionLikeArgs = [],
                functionLikeBody = Body' (Primitive' (Prim' ("Michelson" :| ["add"]) ()) ()) (),
                annLike = ()
              }
          )
          ()
      )
      (),
    Function'
      ( Func'
          ( Like'
              { functionLikedName = "pair",
                functionLikeArgs = [],
                functionLikeBody = Body' (Primitive' (Prim' ("Michelson" :| ["pair"]) ()) ()) (),
                annLike = ()
              }
          )
          ()
      )
      (),
    Type'
      ( Typ'
          { typeUsage = Nothing,
            typeName' = "Storage",
            typeArgs = [],
            typeForm =
              NonArrowed'
                { dataAdt =
                    Product'
                      ( Record'
                          ( Record'''
                              { recordFields =
                                  NameType''
                                    { nameTypeSignature =
                                        Name' ("nat" :| []) (),
                                      nameTypeName = Concrete' "value" (),
                                      annNameType' = ()
                                    }
                                    :| [ NameType''
                                           { nameTypeSignature = Name' ("nat" :| []) (),
                                             nameTypeName = Concrete' "valu" (),
                                             annNameType' = ()
                                           }
                                       ],
                                recordFamilySignature = Nothing,
                                annRecord'' = ()
                              }
                          )
                          ()
                      )
                      (),
                  annNonArrowed = ()
                },
            annTyp = ()
          }
      )
      (),
    Signature'
      ( Sig'
          { signatureName = "foo",
            signatureUsage = Nothing,
            signatureArrowType =
              Infix'
                ( Inf'
                    { infixLeft = Name' ("nat" :| []) (),
                      infixOp = "->" :| [],
                      infixRight =
                        Infix'
                          ( Inf'
                              { infixLeft = Name' ("storage" :| []) (),
                                infixOp = "->" :| [],
                                infixRight = Name' ("storage" :| []) (),
                                annInf = ()
                              }
                          )
                          (),
                      annInf = ()
                    }
                )
                (),
            signatureConstraints = [],
            annSig = ()
          }
      )
      (),
    Function'
      ( Func'
          ( Like'
              { functionLikedName = "foo",
                functionLikeArgs =
                  [ ConcreteA'
                      ( MatchLogic'
                          { matchLogicContents = MatchName' "va" (),
                            matchLogicNamed = Nothing,
                            annMatchLogic = ()
                          }
                      )
                      (),
                    ConcreteA'
                      ( MatchLogic'
                          { matchLogicContents =
                              MatchRecord'
                                ( Punned' ("value" :| []) ()
                                    :| [Punned' ("valu" :| []) ()]
                                )
                                (),
                            matchLogicNamed = Nothing,
                            annMatchLogic = ()
                          }
                      )
                      ()
                  ],
                functionLikeBody =
                  Body'
                    ( ExpRecord'
                        ( ExpressionRecord'
                            { expRecordFields =
                                NonPunned'
                                  ("value" :| [])
                                  ( Application'
                                      ( App'
                                          { applicationName = Name' ("add" :| []) (),
                                            applicationArgs =
                                              Name' ("valu" :| []) ()
                                                :| [Name' ("va" :| []) ()],
                                            annApp = ()
                                          }
                                      )
                                      ()
                                  )
                                  ()
                                  :| [ NonPunned'
                                         ("valu" :| [])
                                         ( Application'
                                             ( App'
                                                 { applicationName = Name' ("add" :| []) (),
                                                   applicationArgs = Name' ("value" :| []) () :| [Name' ("va" :| []) ()],
                                                   annApp = ()
                                                 }
                                             )
                                             ()
                                         )
                                         ()
                                     ],
                              annExpressionRecord = ()
                            }
                        )
                        ()
                    )
                    (),
                annLike = ()
              }
          )
          ()
      )
      (),
    Signature'
      ( Sig'
          { signatureName = "add-storage",
            signatureUsage = Nothing,
            signatureArrowType =
              Infix'
                ( Inf'
                    { infixLeft = Name' ("storage" :| []) (),
                      infixOp = "->" :| [],
                      infixRight = Name' ("nat" :| []) (),
                      annInf = ()
                    }
                )
                (),
            signatureConstraints = [],
            annSig = ()
          }
      )
      (),
    Function'
      ( Func'
          ( Like'
              { functionLikedName = "add-storage",
                functionLikeArgs =
                  [ ConcreteA'
                      ( MatchLogic'
                          { matchLogicContents =
                              MatchRecord'
                                ( Punned' ("value" :| []) ()
                                    :| [Punned' ("valu" :| []) ()]
                                )
                                (),
                            matchLogicNamed = Nothing,
                            annMatchLogic = ()
                          }
                      )
                      ()
                  ],
                functionLikeBody =
                  Body'
                    ( Application'
                        ( App'
                            { applicationName = Name' ("add" :| []) (),
                              applicationArgs = Name' ("value" :| []) () :| [Name' ("valu" :| []) ()],
                              annApp = ()
                            }
                        )
                        ()
                    )
                    (),
                annLike = ()
              }
          )
          ()
      )
      (),
    Signature'
      ( Sig'
          { signatureName = "default",
            signatureUsage = Nothing,
            signatureArrowType =
              Infix'
                ( Inf'
                    { infixLeft = Name' ("nat" :| []) (),
                      infixOp = "->" :| [],
                      infixRight =
                        Infix'
                          ( Inf'
                              { infixLeft = Name' ("storage" :| []) (),
                                infixOp = "->" :| [],
                                infixRight = Name' ("storage" :| []) (),
                                annInf = ()
                              }
                          )
                          (),
                      annInf = ()
                    }
                )
                (),
            signatureConstraints = [],
            annSig = ()
          }
      )
      (),
    Function'
      ( Func'
          ( Like'
              { functionLikedName = "default",
                functionLikeArgs =
                  [ ConcreteA'
                      ( MatchLogic'
                          { matchLogicContents = MatchName' "addnat" (),
                            matchLogicNamed = Nothing,
                            annMatchLogic = ()
                          }
                      )
                      (),
                    ConcreteA'
                      ( MatchLogic'
                          { matchLogicContents = MatchName' "storage" (),
                            matchLogicNamed = Nothing,
                            annMatchLogic = ()
                          }
                      )
                      ()
                  ],
                functionLikeBody =
                  Body'
                    ( Application'
                        ( App'
                            { applicationName = Name' ("pair" :| []) (),
                              applicationArgs =
                                Parened'
                                  ( Application'
                                      ( App'
                                          { applicationName = Name' ("add-storage" :| []) (),
                                            applicationArgs = Name' ("storage" :| []) () :| [],
                                            annApp = ()
                                          }
                                      )
                                      ()
                                  )
                                  ()
                                  :| [ Parened'
                                         ( Application'
                                             ( App'
                                                 { applicationName = Name' ("foo" :| []) (),
                                                   applicationArgs = Name' ("input" :| []) () :| [Name' ("storage" :| []) ()],
                                                   annApp = ()
                                                 }
                                             )
                                             ()
                                         )
                                         ()
                                     ],
                              annApp = ()
                            }
                        )
                        ()
                    )
                    (),
                annLike = ()
              }
          )
          ()
      )
      ()
  ]
